reviews/views.py

```diff
@@ -77,6 +77,83 @@ def get_context_data(self, **kwargs):
         context['data_url'] = reverse("reviews:api:open", args=[self.committee])
 
         return context
+    
+
+class CommitteeMembersWorkloadView(GroupRequiredMixin, CommitteeMixin, generic.TemplateView):
+    #TODO: Look into improving the should_end_date variable. See notes
+
+    template_name = 'reviews/committee_members_workload.html'
+    group_required = [settings.GROUP_SECRETARY]
+    
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context['decisions'] = self.get_all_open_decisions()
+        context['today'] = date.today()
+        context['counts_dict'] = self.get_review_counts_last_year
+
+        return context
+
+    def get_all_open_decisions(self):
+        '''Returns a queryset with all open decisions'''
+
+        objects = Decision.objects.filter(
+            # This fetches all Decisions which are not approved or need revision
+            go__exact = '',
+            review__proposal__reviewing_committee = self.committee,
+            ).select_related(
+            'reviewer',
+            'review',
+            'review__proposal',
+            ).order_by(
+            "reviewer",
+            "review__date_start"
+            )
+
+        return objects
+    
+    def get_review_counts_last_year(self):
+        '''This method creates a dictionary with an overview of all reviews of
+        the past year for each reviewer, by trajectory type (short route,
+        long route, or revision).'''
+
+        dict={}
+        one_year_ago = date.today() - timedelta(days=365)
+
+        decisions_last_year = Decision.objects.filter(
+        review__date_start__gt = one_year_ago,
+        ).select_related(
+        'review',
+        'reviewer',
+        'review__proposal',
+        )
+
+        reviewers = User.objects.exclude(groups=None)
+
+        for reviewer in reviewers:
+            reviewer_full_name = f'{reviewer.first_name} {reviewer.last_name}'
+            decisions_per_reviewer = decisions_last_year.filter(
+            reviewer=reviewer
+            )
+
+            all_decisions_count = len(decisions_per_reviewer)
+
+            all_revisions = decisions_per_reviewer.filter(
+                review__proposal__is_revision = True
+                )
+
+            all_short_routes = decisions_per_reviewer.exclude(
+                review__proposal__is_revision = True
+                ).filter(
+                review__short_route = True
+                )
+
+            revision_counts, short_route_counts = len(all_revisions), len(all_short_routes)
+            long_route_counts = all_decisions_count - revision_counts - short_route_counts
+
+            dict[reviewer_full_name] = [short_route_counts, long_route_counts, revision_counts]
```

Comment:
First off, a small thing:
``reviewer_full_name = f'{reviewer.first_name} {reviewer.last_name}'``
This line re-implements a method we already have: ``user.get_full_name()``. That one is slightly better, so it's probably better to use that instead :)

Okay, now onto the efficiency of the code. Like you mentioned, the current code works but is not the most performant or memory efficient. (Although, it's not that bad either).
I have two different solutions for you, one very simple and one more complex but also even more efficient.

## Simple solution

This one is very simple: just use `qs.count()`. This will do the actual counting in the DB, which it can do a lot more efficiently than doing this in Python. 

Thus, the code would look like this:
```python
for reviewer in reviewers:
    decisions_per_reviewer = decisions_last_year.filter(
        reviewer=reviewer
    )
    
    all_decisions_count = decisions_per_reviewer.count()
    
    revision_count = decisions_per_reviewer.filter(
        review__proposal__is_revision=True
    ).count()
    
    short_route_count = decisions_per_reviewer.exclude(
        review__proposal__is_revision=True
    ).filter(
        review__short_route=True
    ).count()
    
    long_route_counts = all_decisions_count - revision_count - short_route_count
    
    dict[reviewer.get_full_name()] = [
        short_route_count, 
        long_route_count, 
        revision_counts
    ]
```

## Complex solution
The previous code still does 3 DB queries per reviewer, which is probably fine, but we can do better! In many cases, when doing a DB query it takes more time setting up the connection (overhead) than to do the actual query. Thus, having many small DB queries can take a lot more time than one more complex query. In addition, a DB is a lot more optimized for this stuff and can do the same work faster than Python ;)

Note: this is more of a 'hey this exists' portion of the comment than a 'you should use this' thing. It's pretty advanced DB stuff, so I would not be offended if you skip this ;) The previous code is a perfectly fine solution here.

Django has a way to let the DB do the heavy lifting and 'add' data to a object it retrieves using the `qs.annotate()` method. The result is then added as an attribute to the retrieved model. 
Thus, we can use this to calculate all the totals we need in just 1(!) database query. 

A basic example:
```python
User.objects.all().annotate(
   num_decisions=Count("decision")
)
```
Then, with a resulting `User` object we can retrieve the amount of decisions by simply accessing the new `num_decisions` attribute. (`user.num_decisions`)

However, this counts all the decisions, but we want counts for a specific set of decisions. This can be done using the ``filter`` kwarg:
```python
one_year_ago = timezone.now() - timedelta(days=365)

User.objects.all().annotate(
   num_short_route=Count("decision", filter=Q(
      decision__review__date_start__gt=one_year_ago,
      decision__review__stage__gt=Review.SUPERVISOR,
      decision__review__proposal__is_revision=False,
      decision__review__short_route=True
   ))
)
```
Now we do a filtered count for all short route decisions! We can now scale this up for the other values we want:
```python
reviewers = User.objects.filter(groups==self.committee)
one_year_ago = timezone.now() - timedelta(days=365)
base_filter = Q(
    decision__review__date_start__gt=one_year_ago,
    decision__review__stage__gt=Review.SUPERVISOR,
)
return reviewers.annotate(
    total=Count("decision", filter=base_filter),
    num_short_route=Count("decision", filter=base_filter & Q(
        decision__review__proposal__is_revision=False,
        decision__review__short_route=True
    )),
    num_long_route=Count("decision", filter=base_filter & Q(
        decision__review__proposal__is_revision=False,
        decision__review__short_route=False
    )),
    num_revision=Count("decision", filter=base_filter & Q(
        decision__review__date_start__gt=one_year_ago,
        decision__review__proposal__is_revision=True
    )),
)
```
Now we have one DB query which will return all reviewers for a given committee, which will also calculate the decision totals in the DB and add them as attributes. 

Note: I extracted the supervisor/year filters into a separate `Q` and then combined them with the more specific filters using the `&` operator. This makes the code a bit more readable imo. 

Again, this is a 'hey this exists' thing. In this case, the added efficiency is probably not going to be a game changer as the amount of data handled here is small. 
However, if you ever need to work on larger datasets, this becomes very important. (I've seen code runtime go from 30 seconds to just half a second when moving the calculation from python code to an `annotate` query)